From b7321bca8a5969b70bc9338956edc9312ef4cebd Mon Sep 17 00:00:00 2001
From: Roy Pledge <Roy.Pledge@freescale.com>
Date: Mon, 31 Mar 2014 13:49:28 -0400
Subject: [PATCH 631/987] Fix USDPAA DMA Mem freespace calculation

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

The USDPAA DMA freespace calculation should only count
segments with a reference count of zero as free

Signed-off-by: Roy Pledge <Roy.Pledge@freescale.com>
Change-Id: Ib8752f8ad6cb4179b371b6480f47e84033260efe
Reviewed-on: http://git.am.freescale.net:8181/10526
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Geoff Thorpe <Geoff.Thorpe@freescale.com>
Reviewed-by: Jose Rivera <German.Rivera@freescale.com>
---
 drivers/staging/fsl_qbman/fsl_usdpaa.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/staging/fsl_qbman/fsl_usdpaa.c b/drivers/staging/fsl_qbman/fsl_usdpaa.c
index 6788582..021dff8 100644
--- a/drivers/staging/fsl_qbman/fsl_usdpaa.c
+++ b/drivers/staging/fsl_qbman/fsl_usdpaa.c
@@ -1053,7 +1053,8 @@ static long ioctl_dma_stats(struct ctx *ctx, void __user *arg)
 	result.total_bytes = phys_size;
 
 	list_for_each_entry(frag, &mem_list, list) {
-		result.free_bytes += frag->len;
+		if (frag->refs == 0)
+			result.free_bytes += frag->len;
 	}
 
 	return copy_to_user(arg, &result, sizeof(result)); }
-- 
1.9.1

